### PR TITLE
SCP-5011 Update `marlowe-web-server` to use `marlowe-proxy`

### PR DESCRIPTION
**Pull request by jhbertra**
Fairly straightforward - most file changes are related to pulling out a separate cabal project for the web server.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
    - [x] Reviewer requested

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -214,6 +214,7 @@ Services currently included:
 * `marlowe--sync`: `marlowe-sync` for the `preprod` network.
 * `marlowe--indexer`: `marlowe-indexer` for the `preprod` network.
```
Comment (bwbush) [inline README.adoc:215]: Change `--` to `-` in lines 214 and 215.

File: deploy/operables.nix [hunk 2]
```diff
@@ -346,24 +346,18 @@ in
...
       ${packages.marlowe-web-server}/bin/marlowe-web-server \
         --http-port "$PORT" \
-        --marlowe-sync-host "$SYNC_HOST" \
-        --marlowe-query-port "$MARLOWE_QUERY_PORT" \
-        --tx-host "$TX_HOST" \
-        --tx-command-port "$TX_PORT" \
+        --marlowe-runtime-host "$RUNTIME_HOST" \
+        --marlowe-runtime-port "$RUNTIME_PORT" \
         --enable-open-api \
         --access-control-allow-origin-all
```
Comment (bwbush) [inline deploy/operables.nix:362]: Could we include the bind address as an option? Are `127.0.0.1` and `0.0.0.0` the only possibilities currently?

File: marlowe-runtime-web/NOTICE [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Copyright 2019 Input Output (Hong Kong) Ltd.
```
Comment (bwbush) [inline marlowe-runtime-web/NOTICE:1]: Technically, 2022.

File: scripts/local-docker.env [hunk 4]
```diff
@@ -3,14 +3,6 @@ export MARLOWE_CHAIN_SYNC_PORT=$(docker-compose port marlowe-chain-sync 3715 | s
 export MARLOWE_CHAIN_SYNC_QUERY_PORT=$(docker-compose port marlowe-chain-sync 3716 | sed -e s/://)
 export MARLOWE_CHAIN_SYNC_COMMAND_PORT=$(docker-compose port marlowe-chain-sync 3720 | sed -e s/://)
 
-export MARLOWE_RT_SYNC_HOST=127.0.0.1
```
Comment (bwbush) [inline scripts/local-docker.env:6]: Do we really want to completely hide these? I'm wondering if there are some use cases where a client might want to use a service directly.

**Review by bwbush — APPROVED** [review]
I left a few minor comments unrelated to the code itself. I haven't tested deploying this, so let me know if you'd like me to.

File: deploy/operables.nix (hunk 2 above)
Comment (jhbertra) [reply deploy/operables.nix:362]: Yeah, we can do this, but I'd rather do it in a separate PR

File: scripts/local-docker.env (hunk 4 above)
Comment (jhbertra) [reply scripts/local-docker.env:6]: The ports are still exposed, they would just need to be looked up manually (I don't want to encourage it)

File: README.adoc (hunk 1 above)
Comment (jhbertra) [reply README.adoc:215]: I'll add this change to my next PR

File: marlowe-runtime-web/NOTICE (hunk 3 above)
Comment (jhbertra) [reply marlowe-runtime-web/NOTICE:1]: I'll do a round of fixing these in the next PR

**Pull request merged**